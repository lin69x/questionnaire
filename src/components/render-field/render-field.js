import React from 'react';
import { Field } from 'redux-form';

const styles = {
  radioLabel: {
    cursor: 'pointer',
  }
};

export const renderRadioGroup = ({ label, options, name}) => (
  <div>
    <h4>{label}</h4>
    {
      options.map((option, key) => (
        <React.Fragment key={key}>
          <label style={styles.radioLabel}>
            <Field
              name={name}
              component="input"
              type="radio"
              value={option.value}
            />{' '}
            {option.label}
          </label>
          <br/>
        </React.Fragment>
      ))
    }
  </div>
);

export const renderInputField = ({ label, name, textarea=false, ...rest }) => (
  <div>
    <h4>{label}</h4>
    <Field
      name={name}
      component={ textarea ? 'textarea' : 'input'}
      type="text"
      {...rest}
    />
  </div>
);
