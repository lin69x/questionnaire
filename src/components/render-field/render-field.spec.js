import React from 'react';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {reduxForm} from 'redux-form';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import {renderRadioGroup, renderInputField} from './render-field';

Enzyme.configure({adapter: new EnzymeAdapter()});

const spy = jest.fn();
const store = createStore(() => ({}));

describe('render radio group', () => {
  it('RADIO GROUP should match snapshot', () => {
    const radioForm = () => (
      <form>
        {
          renderRadioGroup({
            label: "Radio Group",
            name: 'testRadio',
            options: [
              {
                label: 'option 1',
                value: 'option 1',
              },
              {
                label: 'option 2',
                value: 'option 2',
              }
            ]
          })
        }
      </form>
    );

    const Decorated = reduxForm({form: 'testRadioForm'})(radioForm);

    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <Decorated
            handleSubmit={spy}
            submitting={false}
            submit={spy}
          />
        </Provider>
      ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('INPUT FIELD should match snapshot', () => {
    const inputForm = () => (
      <form>
        {
          renderInputField({
            label: 'input field',
            name: 'testInput'
          })
        }
      </form>
    );

    const Decorated = reduxForm({form: 'testInputForm'})(inputForm);

    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <Decorated
            handleSubmit={spy}
            submitting={false}
            submit={spy}
          />
        </Provider>
      ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('TEXTAREA should match snapshot', () => {
    const textAreaForm = () => (
      <form>
        {
          renderInputField({
            label: 'input field',
            name: 'testInput',
            textarea: true,
          })
        }
      </form>
    );

    const Decorated = reduxForm({form: 'testTextAreaForm'})(textAreaForm);

    const renderedValue = renderer
      .create(
        <Provider store={store}>
          <Decorated
            handleSubmit={spy}
            submitting={false}
            submit={spy}
          />
        </Provider>
      ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
