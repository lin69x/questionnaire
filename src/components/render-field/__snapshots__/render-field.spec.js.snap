// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render radio group INPUT FIELD should match snapshot 1`] = `
<form>
  <div>
    <h4>
      input field
    </h4>
    <input
      name="testInput"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;

exports[`render radio group RADIO GROUP should match snapshot 1`] = `
<form>
  <div>
    <h4>
      Radio Group
    </h4>
    <label
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <input
        checked={false}
        name="testRadio"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="radio"
        value="option 1"
      />
       
      option 1
    </label>
    <br />
    <label
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <input
        checked={false}
        name="testRadio"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="radio"
        value="option 2"
      />
       
      option 2
    </label>
    <br />
  </div>
</form>
`;

exports[`render radio group TEXTAREA should match snapshot 1`] = `
<form>
  <div>
    <h4>
      input field
    </h4>
    <textarea
      name="testInput"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
</form>
`;
